pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/service/api/Component.java

```diff
@@ -144,13 +158,34 @@ void uploadFunction(final InputStream uploadedInputStream,
 
     StreamingOutput downloadFunction(String path,
                                      String clientRole,
-                                     AuthenticationDataHttps clientAuthenticationDataHttps);
+                                     final AuthenticationDataSource clientAuthenticationDataHttps);
+
+    @Deprecated
+    default StreamingOutput downloadFunction(String path,
```

Comment:
Set this method to deprecated state, in order to maintain backward compatibility